firedrake/plot.py

```diff
@@ -129,10 +128,10 @@ def bezier_plot(function, axes=None, **kwargs):
     M_inv = np.linalg.inv(M)
     cell_node_list = function.function_space().cell_node_list
     y_vals = np.dot(function.dat.data_ro[cell_node_list], M_inv)
-    x = SpatialCoordinate(function.function_space().mesh())
-    coords = Function(function.function_space())
+    x = SpatialCoordinate(mesh)
+    coords = Function(FunctionSpace(mesh, 'DG', deg))
     coords.interpolate(x[0])
-    x_vals = np.dot(coords.dat.data_ro[cell_node_list], M_inv)
+    x_vals = np.dot(coords.dat.data_ro[coords.function_space().cell_node_list], M_inv)
```

Comment:
Taylor basis is DG and doesn't have the Lagrange basis.

--cjc

On 29 July 2016 at 11:21, Zhou Fangyi <notifications@github.com> wrote:

> In firedrake/plot.py
> <https://github.com/firedrakeproject/firedrake/pull/845#discussion_r72771658>
> :
>
> >      coords.interpolate(x[0])
> > -    x_vals = np.dot(coords.dat.data_ro[cell_node_list], M_inv)
> > +    x_vals = np.dot(coords.dat.data_ro[coords.function_space().cell_node_list], M_inv)
>
> To be honest, I am not sure about this. The assumption is that DG and CG
> have the same basis at the same given degree.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/firedrakeproject/firedrake/pull/845/files/7cbad3bcaca2adc4c26218ef74d9fee6500e2e50#r72771658>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF0q8hQhKf48pQQD9e1ZzMECJxpMeJctks5qadQMgaJpZM4JYDMv>
> .
>



-- 
http://www.imperial.ac.uk/people/colin.cotter

www.cambridge.org/9781107663916